lib/deref/index.js

```diff
@@ -1,65 +1,62 @@
-var __key = require("./../internal/key");
-var __path = require("./../internal/path");
-var __parent = require("./../internal/parent");
 var InvalidDerefInputError = require("./../errors/InvalidDerefInputError");
+var getCachePosition = require("./../get/getCachePosition");
+var CONTAINER_DOES_NOT_EXIST = "e";
 
 module.exports = function deref(boundJSONArg) {
-    var path = boundJSONArg && boundJSONArg[__path];
 
-    // The simple deref when there is a path.
-    if (path) {
-        return this._clone({
-            _path: path
-        });
-    }
-
-    var key = boundJSONArg && boundJSONArg[__key];
-
-    // This is an invalid argument.  Must* be a generated JSON object from
-    // get.  * - Technically this can just be passed in but don't do that.
-    if (path === undefined && key === undefined) {
-        throw new InvalidDerefInputError();
-    }
-
-    // We have to follow the path back up recursively until we hit a path
-    // or a null parent.
-    var reversedKeys = [key];
-    var reversedLength = 0;
-    var current = boundJSONArg[__parent];
-    path = null;
-    while (current !== null && !path) {
-        key = current[__key];
-        path = current[__path];
+    var absolutePath = boundJSONArg && boundJSONArg.$__path;
```

Comment:
Did we decide to stop using the internal/prefix? Just want to make sure it's conscious, since I don't remember it being part of the __path obfuscation discussion.